### ISSUE TITLE
POST requests being cached

### ISSUE THREAD

**Issue by nicolasgarnil** [typhoeus/typhoeus#479]
I have configured the following caching strategy in the configuration file:

``` ruby
class Cache
  def get(request)
    Rails.cache.read(request)
  end

  def set(request, response)
    Rails.cache.write(request, response)
  end
end
```

``` ruby
Typhoeus::Config.cache = Cache.new
```

It works as expected for GET requests but I have just noticed it is also caching POST requests. Is this the default behaviour? How can I disable the cache for POST/PUT requests?

**Comment by Zapotek** [comment]
Why not do the check in the cache?
Seems to me like the Typhoeus cache is application level, not protocol level, it doesn't care whether or not a request is idempotent or not.

**Comment by nicolasgarnil** [comment]
I agree. Although in most cases you will not want to cache POST request it is better to let the developer take that decision.

This workaround did the trick:

```ruby
class Cache
  def get(request)
    Rails.cache.read(request)
  end

  def set(request, response)
    return if request.headers['Cache-Control'] =~ /no-cache|max-age=0/
    Rails.cache.write(request, response)
  end
end
```

**Comment by felipedmesquita** [comment]
Thanks @nicolasgarnil for the report. Since you've solved your issue and Typhoeus was working as expected, I'm closing this.
